src/main/java/org/embulk/input/gcs/GcsFileInput.java

```diff
@@ -91,28 +91,50 @@ static FileList listFiles(final PluginTask task) {
 
     // String nextToken = base64Encode(0x0a + ASCII character according to utf8EncodeLength position+ filePath);
     static String base64Encode(final String path) {
+        byte[] lengthVarint;
         byte[] encoding;
         byte[] utf8 = path.getBytes(StandardCharsets.UTF_8);
         LOG.debug("path string: {} ,path length:{} \" + ", path, utf8.length);
 
         int utf8EncodeLength = utf8.length;
-        if (utf8EncodeLength >= 128) {
+        if (utf8EncodeLength >= 65_535) {
             throw new ConfigException(String.format("last_path '%s' is too long to encode. Please try to reduce its length", path));
         }
 
-        encoding = new byte[utf8.length + 2];
+        lengthVarint = encodeVarint(utf8EncodeLength);
+        encoding = new byte[1 + lengthVarint.length + utf8.length];
         encoding[0] = 0x0a;
 
-        // for example: 60 -> '<'
-        char temp = (char) utf8EncodeLength;
-        encoding[1] = (byte) temp;
-        System.arraycopy(utf8, 0, encoding, 2, utf8.length);
+        System.arraycopy(lengthVarint, 0, encoding, 1, lengthVarint.length);
+        System.arraycopy(utf8, 0, encoding, 1 + lengthVarint.length, utf8.length);
 
         final String s = Base64.getEncoder().encodeToString(encoding);
         LOG.debug("last_path(base64 encoded): {}", s);
         return s;
     }
 
+    // see: https://protobuf.dev/programming-guides/encoding/#varints
+    private static byte[] encodeVarint(int value)
```

Comment:
I understand that this method is indirectly tested via `base64Encode`, but for sure and for readability, can you add some more tests directly for this `encodeVarint` method?

Reply:
Tests were added. I use reflect to test private methods, but please point out if this should not be used as repository policy.
https://github.com/embulk/embulk-input-gcs/pull/72/commits/9f691e6b491af347542c3e097feb521cc9fd3eb7

Reply:
Thanks for your consideration! I think that the method can be just package-private (no visibility modifier) so that the method is visible from test classes.

Reply:
Fixed to package-private. Thanks for the comment!
https://github.com/embulk/embulk-input-gcs/pull/72/commits/dea553bc2a2b87cdbe782c17391419ab713b9b60